src/util/token.ts

```diff
@@ -1,15 +1,27 @@
-import {get} from "lodash";
-import {WrappedDocument, getData} from "@govtechsg/open-attestation";
+import {getData, v2, v3, WrappedDocument} from "@govtechsg/open-attestation";
 
-import {Issuer} from "../types";
+export const getBatchMerkleRoot = (
+  document: WrappedDocument<v2.OpenAttestationDocument> | WrappedDocument<v3.OpenAttestationDocument>
+) => {
+  return `0x${document.signature?.merkleRoot || ""}`;
+};
 
-export const getBatchMerkleRoot = (document: WrappedDocument) => {
-  return `0x${get(document, "signature.merkleRoot")}`;
+const isWrappedV3Document = (document: any): document is WrappedDocument<v3.OpenAttestationDocument> => {
+  return document && document.version === "open-attestation/3.0";
+};
+const isWrappedV2Document = (document: any): document is WrappedDocument<v2.OpenAttestationDocument> => {
+  return !isWrappedV3Document(document);
 };
 
-export const getIssuer = (document: WrappedDocument): Issuer => {
+export const getIssuerAddress = (
```

Comment:
wonder if everything here should live in open-attestation? no need to do it right now just wondering

Reply:
yeah agree might be helpful on open-attestation